src/main/scala/is/hail/expr/ir/functions/SetFunctions.scala

```diff
@@ -69,5 +69,66 @@ object SetFunctions extends RegistryFunctions {
         ApplySpecial("&&",
           FastSeq(Ref(a, TBoolean()), SetContains(w, Ref(x, t)))))
     }
+
+    registerIR("min", TSet(tnum("T"))) { s =>
+      val t = -s.typ.asInstanceOf[TSet].elementType
+      val a = genUID()
+      Let(a, ToArray(s), If(
+        ApplySpecial(">", FastSeq(ArrayLen(Ref(a, TArray(t))), I32(0))),
+        ArrayRef(Ref(a, TArray(t)), I32(0)),
+        NA(t)))
+    }
+
+    registerIR("max", TSet(tnum("T"))) { s =>
+      val t = -s.typ.asInstanceOf[TSet].elementType
```

Comment:
the type of `Ref(a)` is not quite going to be `TArray(t)`, since ToArray is `TArray(s.typ.elementType)`. Here and below in median.